src/main/java/cc/reconnected/kromer/Kromer.java

```diff
@@ -160,65 +162,83 @@ public void onInitializeServer() {
         );
     }
 
-    public static void executeWelfare() {
+    private static float calculateWelfare(){
         List<ServerPlayer> playersWithSupporter = client.server
-            .getPlayerList()
-            .getPlayers()
-            .stream()
-            .filter(z -> Permissions.check(z, config.SupporterGroup()) && !Permissions.check(z, config.BotPermission()))
-            .toList();
+                .getPlayerList()
+                .getPlayers()
+                .stream()
+                .filter(z -> Permissions.check(z, config.SupporterGroup()) && !Permissions.check(z, config.BotPermission()))
+                .toList();
 
         List<ServerPlayer> playersWithoutSupporter = client.server
-            .getPlayerList()
-            .getPlayers()
-            .stream()
-            .filter(z -> !Permissions.check(z, config.SupporterGroup()) && !Permissions.check(z, config.BotPermission()))
-            .toList();
+                .getPlayerList()
+                .getPlayers()
+                .stream()
+                .filter(z -> !Permissions.check(z, config.SupporterGroup()) && !Permissions.check(z, config.BotPermission()))
+                .toList();
 
         float welfare = config.HourlyWelfare();
         if (
-            !playersWithSupporter.isEmpty() &&
-            !playersWithoutSupporter.isEmpty()
+                !playersWithSupporter.isEmpty() &&
+                !playersWithoutSupporter.isEmpty()
         ) {
             welfare =
-                config.HourlyWelfare() *
-                (config.SupporterMultiplier() * playersWithSupporter.size());
+                    config.HourlyWelfare() *
+                    (config.SupporterMultiplier() * playersWithSupporter.size());
         }
+        return welfare;
+    }
 
-        float finalWelfare = Math.round(welfare * 100f) / 100f;
+    public static void executeWelfare() {
+
+        float finalWelfare = calculateWelfare();
 
         client.server
             .getPlayerList()
             .getPlayers()
             .forEach(p -> {
-                if(config.BotPermission() != null) {
-                    if(Permissions.check(p, config.BotPermission())) return; // If bot then bye
-                }
-
-                Wallet wallet = Kromer.database.getWallet(p.getUUID());
-                if (wallet == null) return;
-
-                if (
-                    Solstice.modules.getModule(AfkModule.class).get().isPlayerAfk(p)
-                ) return;
```

Comment:
Hey, I'm pretty sure this wasn't intentional, but the whole AFK player check for welfare is gone in this.